raster/src/main/scala/geotrellis/raster/io/geotiff/ByteGeoTiffSegment.scala

```diff
@@ -9,13 +9,22 @@ import java.util.BitSet
 import spire.syntax.cfor._
 
 
+class NoDataByteGeoTiffSegment(bytes: Array[Byte], noDataValue: Byte) extends ByteGeoTiffSegment(bytes) {
+  override
+  def get(i: Int): Int =
+    if (bytes(i) == noDataValue)
+      NODATA
+    else
+      bytes(i)
+
+}
+
 class ByteGeoTiffSegment(val bytes: Array[Byte]) extends GeoTiffSegment {
   val size: Int = bytes.size
 
   def getInt(i: Int): Int = get(i)
   def getDouble(i: Int): Double = i2d(get(i))
-
-  def get(i: Int): Int = bytes(i) & 0xFF
+  def get(i: Int): Int = bytes(i)
```

Comment:
Wondering why this is different from https://github.com/geotrellis/geotrellis/blob/320072b10d64387ca1945bb42ad47f69ce28611b/raster/src/main/scala/geotrellis/raster/io/geotiff/ByteGeoTiffSegment.scala#L26
Can you think of a reason that the `get` should return an `Int`?